### ISSUE TITLE
option -F / --filesize creates much larger files than mydumper

### ISSUE THREAD

**Issue by nullnotnil** [pingcap/dumpling#143]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

When using the option `-F`, I noticed the filesizes were double what I requested. So I checked mydumper, and it seems to be able to estimate file size correctly.

The workaround is easy (I specify a value of 50% of what I expect). But it should be possible to use a similar algorithm to mydumper.

2. What did you expect to see?

Dumping the ontime database, using mydumper from distro repos:

```
mydumper -P 4000 --database ontime -F 256 --no-locks

-rw-rw-r-- 1 nullnotnil nullnotnil 246M Aug 24 08:14 ontime.ontime.00001.sql
-rw-rw-r-- 1 nullnotnil nullnotnil 246M Aug 24 08:14 ontime.ontime.00002.sql
```

3. What did you see instead?

Comparing to dumpling:

```
dumpling --database ontime -o ontime-dumpling -F 256MiB

-rwxr-xr-x 1 nullnotnil nullnotnil 514M Aug 24 08:15 ontime.ontime.0.sql
-rwxr-xr-x 1 nullnotnil nullnotnil 514M Aug 24 08:15 ontime.ontime.1.sql
-rwxr-xr-x 1 nullnotnil nullnotnil 514M Aug 24 08:15 ontime.ontime.2.sql

dumpling --database ontime -o ontime-dumpling -F 128MiB

-rwxr-xr-x 1 nullnotnil nullnotnil 257M Aug 24 08:16 ontime.ontime.0.sql
-rwxr-xr-x 1 nullnotnil nullnotnil 257M Aug 24 08:16 ontime.ontime.1.sql
-rwxr-xr-x 1 nullnotnil nullnotnil 257M Aug 24 08:16 ontime.ontime.2.sql
-rwxr-xr-x 1 nullnotnil nullnotnil 257M Aug 24 08:16 ontime.ontime.3.sql

/mnt/evo970/bin/dumpling --database ontime -o ontime-dumpling -F 512MiB

-rwxr-xr-x 1 nullnotnil nullnotnil 1.1G Aug 24 08:18 ontime.ontime.0.sql
-rwxr-xr-x 1 nullnotnil nullnotnil 1.1G Aug 24 08:18 ontime.ontime.1.sql
-rwxr-xr-x 1 nullnotnil nullnotnil 1.1G Aug 24 08:19 ontime.ontime.2.sql
-rwxr-xr-x 1 nullnotnil nullnotnil 1.1G Aug 24 08:19 ontime.ontime.3.sql
```

4. Versions of the cluster

    - Dumpling version (run `dumpling -V`):

```
Release version: v4.0.3-10-gfd45682
Git commit hash: fd4568290becf03fcc8ee634194bc34ef3f61b53
Git branch:      master
Build timestamp: 2020-08-21 06:26:03Z
Go version:      go version go1.13 linux/amd64
```

    - Source database version (execute `SELECT version();` in a MySQL client):

```
mysql> select tidb_version()\G
*************************** 1. row ***************************
tidb_version(): Release Version: v4.0.0-beta.2-1005-gf8518988e
Edition: Community
Git Commit Hash: f8518988e445058efe240c4c7cbef6ac2f38273b
Git Branch: master
UTC Build Time: 2020-08-21 08:33:34
GoVersion: go1.13
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false
1 row in set (0.01 sec)
```

    - Other interesting information (system version, hardware config, etc):

        >
        >

5. Operation logs
   - Please upload `dumpling.log` for Dumpling if possible
   - Other interesting logs
   
6. Configuration of the cluster and the task
   - running command for Dumpling if possible

**Comment by nullnotnil** [comment]
It's possible this might be a duplicate of #127. I can take a look after it merges.

**Comment by lichunzhu** [comment]
After #127 is solved this problem will be solved. Currently, `dumpling` only counts the length of values when `-F` is enabled.

**Comment by lichunzhu** [comment]
Have been solved by https://github.com/pingcap/dumpling/issues/127. Reopen this if filesize is much larger.
